File: php/class-access/sub/Facade.php
<?php

namespace sub;

require_once(__DIR__ . '/Processor.php');
require_once(__DIR__ . '/Value.php');

// NOTE: どこからでも利用可能
class Facade
{
    private $processor;

    public function __construct()
    {
        $this->processor = new Processor();
    }

    public function execute(int $input): void
    {
        $value = new Value($input);
        $this->processor->process($value);
    }
}

File: java/class-access/sub/Facade.java
package sub;

// NOTE: public class なのでパッケージの外から利用可能
public class Facade {
    private Processor processor;

    public Facade() {
        processor = new Processor();
    }

    public void execute(int input) {
        Value value = new Value(input);
        processor.process(value);
    }
}
